pandas/core/generic.py

```diff
@@ -831,9 +835,11 @@ def _set_axis_name(self, name, axis=0):
         axis = self._get_axis_number(axis)
         idx = self._get_axis(axis).set_names(name)
 
-        renamed = self.copy(deep=True)
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+        renamed = self if inplace else self.copy()
```

Comment:
It's the default and can thus be removed (as of @jreback's request that disappeared after my rebase).
https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L3523